src/sentry/sentry_metrics/querying/api.py

```diff
@@ -121,98 +313,139 @@ def _generate_full_series(
     return full_series
 
 
-def _translate_query_results(
-    interval: int,
-    query_results: Sequence[QueryResult],
-) -> Mapping[str, Any]:
+def _get_identity(value: ResultValue) -> ResultValue:
+    """
+    Computes the identity of a value.
+
+    For nan, we want to return None instead of 0.0 but this is just a design decision that conforms
+    to the previous implementation of the layer.
+    """
+    if value is None:
+        return None
+
+    if _is_nan(value):
+        return None
+
+    # We might decide in the future to have identity values specific to each aggregate.
+    return type(value)()
+
+
+def _nan_to_none(value: ResultValue) -> ResultValue:
+    """
+    Converts a nan value to None or returns the original value.
+    """
+    if value is None:
+        return None
+
+    if _is_nan(value):
+        return None
+
+    return value
+
+
+def _is_nan(value: ResultValue) -> bool:
+    """
+    Returns whether the result of a query is nan.
+    """
+    if value is None:
+        return False
+    elif isinstance(value, list):
+        return any(map(lambda e: e is not None and math.isnan(e), value))
```

Comment:
The better implementation here, would be to mutate any `nan` to `None` in the list but for now we just return `None`.